syntax = "proto3";
package echo;

message EchoRequest
{
  string message = 1;
  optional string aaa = 2;
  map<string, string> values = 3;
};

message EchoResponse
{
  string response = 1;
};

message DummyRequest
{
  string message = 1;
};

message DummyResponse
{
};

service EchoService
{
  rpc Echo(EchoRequest) returns (EchoResponse);
  rpc Dummy(DummyRequest) returns (DummyResponse);
};

option cc_generic_services = true;
